### PR TITLE
docs: update how forking works

### PR DESCRIPTION
**Pull request by mattsse**
* document new cheatcodes introduced in https://github.com/foundry-rs/foundry/pull/2301
* update docs/examples how multifork works

### CODE REVIEW

**Review by 0xvv — APPROVED** [review]
Everything is clear, no comments

File: src/cheatcodes/is-persistent.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+### Signature
+
+```solidity
+ function isPersistent(address) external returns (bool);
+```
+
+### Description
+
+Returns whether an account is marked as persistent (`makePersistent`)[./make-persistent.md].
```
Comment (onbjerg) [inline src/cheatcodes/is-persistent.md:11]:
```suggestion
Returns whether an account is marked as persistent ([`makePersistent`](./make-persistent.md)). 
```

File: src/cheatcodes/is-persistent.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+### Description
+
+Returns whether an account is marked as persistent (`makePersistent`)[./make-persistent.md]. 
+
+### Examples
+
+Check default status of `msg.sender` and the current test account
+
+```solidity
+// by default the `sender` and the contract itself are persistent
```
Comment (onbjerg) [inline src/cheatcodes/is-persistent.md:18]:
```suggestion
// By default the `sender` and the test contract itself are persistent
```

File: src/cheatcodes/make-persistent.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+```solidity
+function makePersistent(address) external;
+function makePersistent(address, address) external;
+function makePersistent(address, address, address) external;
+function makePersistent(address[] calldata) external;
+```
+
+### Description
+
+Each fork (`createFork`)[./create-fork.md] has its own independent storage, which is also replaced when another fork is selected (`selectFork`)[../select-fork.md].
```
Comment (onbjerg) [inline src/cheatcodes/make-persistent.md:14]:
```suggestion
Each fork ([`createFork`](./create-fork.md)) has its own independent storage, which is also replaced when another fork is selected ([`selectFork`](./select-fork.md)).
```

File: src/cheatcodes/make-persistent.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+By default, only the test contract account and the caller are persistent across forks, which means that changes to the state of the test contract (variables) are preserved when different forks are selected. This way data can be shared by storing it in the contract's variables. 
+
+However, with the cheatcode `makePersistent(address)` the specified accounts are marked as persistent, which means that their state is available regardless of which fork is currently active.
+
+### Examples
+
+Mark a new contract as persistent
+
+```solidity
+
```
Comment (onbjerg) [inline src/cheatcodes/make-persistent.md:24]:
```suggestion
```

File: src/cheatcodes/make-persistent.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+function makePersistent(address, address, address) external;
+function makePersistent(address[] calldata) external;
+```
+
+### Description
+
+Each fork (`createFork`)[./create-fork.md] has its own independent storage, which is also replaced when another fork is selected (`selectFork`)[../select-fork.md].
+By default, only the test contract account and the caller are persistent across forks, which means that changes to the state of the test contract (variables) are preserved when different forks are selected. This way data can be shared by storing it in the contract's variables. 
+
+However, with the cheatcode `makePersistent(address)` the specified accounts are marked as persistent, which means that their state is available regardless of which fork is currently active.
```
Comment (onbjerg) [inline src/cheatcodes/make-persistent.md:17]:
```suggestion
However, with this cheatcode, it is possible to mark the specified accounts as persistent, which means that their state is available regardless of which fork is currently active.
```

File: src/cheatcodes/revoke-persistent.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+### Signature
+
+```solidity
+    function revokePersistent(address) external;
+    function revokePersistent(address[] calldata) external;
+```
+
+### Description
+
+The counterpart of (`makePersistent`)[./make-persistent.md], that makes the given contract not persistent across fork swaps
```
Comment (onbjerg) [inline src/cheatcodes/revoke-persistent.md:12]:
```suggestion
The counterpart of [`makePersistent`](./make-persistent.md), that makes the given contract not persistent across fork swaps
```

File: src/cheatcodes/revoke-persistent.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+### Examples
+
+Revoke a persistent status of a contract
+
+```solidity
+
```
Comment (onbjerg) [inline src/cheatcodes/revoke-persistent.md:19]:
```suggestion
```

**Review by onbjerg — APPROVED** [review]
some nits (mostly making links work)

**Pull request merged**